components/filecoin/rs/src/signature.rs

```diff
@@ -0,0 +1,196 @@
+    Secp256k1(#[from] libsecp256k1::Error),
+    // Key decoding error
+    #[error("key decoding error (only hex or base64 is accepted)")]
+    KeyDecoding(),
+    /// InvalidBigInt error
+    #[error("InvalidBigInt error")]
+    InvalidBigInt(#[from] num_bigint_chainsafe::ParseBigIntError),
+    /// Generic error message
+    #[error("Error: `{0}`")]
+    GenericString(String),
+    /// Not able to parse integer
+    #[error("Cannot parse integer")]
+    ParseIntError(#[from] ParseIntError),
+    /// BLS error
+    #[error("BLS error | {0}")]
+    BLS(#[from] bls_signatures::Error),
+    /// Invalid BIP44Path
+    #[error("Invalid BIP44 path : `{0}`")]
+    InvalidBIP44Path(#[from] zx_bip44::errors::BIP44PathError),
+    /// BLS error
+    #[error("Couldn't convert from slice")]
+    TryFromSlice(#[from] TryFromSliceError),
+    /// Base64 decode Error
+    #[error("Base64 decode error | {0}")]
+    DecodeError(#[from] base64::DecodeError),
+
+    #[error("Marshall error | {0}")]
+    FvmSharedEncodingError(#[from] fvm_ipld_encoding::Error),
+}
+
+/// Multihash prefix (https://multiformats.io/multihash/) for blake2b 256 bits
+/// 01 - Version 1
+/// 71 - DAG_CBOR encoded
+/// a0e402 - code for blake2b256 hashing function
+/// 20 - 32 bytes length
+const CID_PREFIX: &[u8] = &[0x01, 0x71, 0xa0, 0xe4, 0x02, 0x20];
+const HASH_LENGTH: usize = 32;
+
+/// transform a message into a hashed message ready to be signed and following Filecoin standard
+/// https://github.com/Zondax/filecoin-signing-tools/blob/2e7b005b4733761de11d4a252cb481ca5aedb029/signer/src/utils.rs#L19
+pub fn get_digest(message: &[u8]) -> Result<[u8; HASH_LENGTH], TryFromSliceError> {
+    let message_hashed =
+        Params::new().hash_length(HASH_LENGTH).to_state().update(message).finalize();
+    // Making blake2b256 CID multihash, https://multiformats.io/multihash/
+    let cid_hashed = Params::new()
+        .hash_length(HASH_LENGTH)
+        .to_state()
+        .update(CID_PREFIX)
+        .update(message_hashed.as_bytes())
+        .finalize();
+
+    cid_hashed.as_bytes().try_into()
+}
+
+
+fn sign_message_secp256(
+  private_key: &PrivateKey, message_digest: &libsecp256k1::Message
+) -> Result <SecpSignature, SignerError> {
+  let secret_key = libsecp256k1::SecretKey::parse_slice(&private_key.0)?;
+  let (signature_rs, recovery_id) = sign(&message_digest, &secret_key);
+  Ok(SecpSignature {signature: signature_rs, recid: recovery_id})
+}
+
+// https://github.com/Zondax/filecoin-signing-tools/blob/2e7b005b4733761de11d4a252cb481ca5aedb029/signer/src/lib.rs#L282
+fn transaction_sign_secp56k1_raw(
+    unsigned_message_api: &UnsignedMessage,
+    private_key: &PrivateKey,
+) -> Result<SignatureSECP256K1, SignerError> {
+    let message_cbor = unsigned_message_api.marshal_cbor()?;
+    let cid_hashed = get_digest(message_cbor.as_ref())?;
+    let message_digest = Message::parse_slice(&cid_hashed)?;
+    let signed = sign_message_secp256(&private_key, &message_digest)?;
```

Comment:
`private_key` is already a reference here, you don't need the `&`.

Reply:
done